File: generator/app/templates/app/src/main/java/com/example/base/baseapp/model/ExampleData.java
package com.example.base.baseapp.model;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import java.util.ArrayList;
import java.util.List;

@Entity
public class ExampleData {

    @NonNull
    @PrimaryKey
    @ColumnInfo(name = "id")
    private String id;

    @ColumnInfo(name = "content")
    private String content;

    private static int NUMBER_OF_ELEMENTS = 10;

    public ExampleData(String id, String content) {
        this.id = id;
        this.content = content;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public static List<ExampleData> populateData() {
        ArrayList<ExampleData> exampleData = new ArrayList<>();
        for (int i = 0; i < NUMBER_OF_ELEMENTS; i++) {
            exampleData.add(new ExampleData(String.valueOf(i), "Data " + i));
        }
        return exampleData;
    }
}

File: generators/app/templates/app/src/main/kotlin/com/example/base/baseapp/model/ExampleData.kt
package com.example.base.baseapp.model

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

import java.util.ArrayList

@Entity
class ExampleData(@field:PrimaryKey
                  @field:ColumnInfo(name = "id")
                  var id: String, @field:ColumnInfo(name = "content")
                  var content: String?) {
    companion object {

        private val NUMBER_OF_ELEMENTS = 10

        fun populateData(): List<ExampleData> {
            val exampleData = ArrayList<ExampleData>()
            for (i in 0 until NUMBER_OF_ELEMENTS) {
                exampleData.add(ExampleData(i.toString(), "Data $i"))
            }
            return exampleData
        }
    }
}
